src/test/java/org/cactoos/io/TempFileTest.java

```diff
@@ -0,0 +1,114 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.cactoos.func.MatcherOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link TempFile}.
+ *
+ * @author George Aristy (george.aristy@gmail.com)
+ * @version $Id$
+ * @since 1.0
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class TempFileTest {
+
+    @Test
+    public void tmpFileIsCreated() throws Exception {
+        try (final TempFile file = new TempFile()) {
+            MatcherAssert.assertThat(
+                "Cannot create a temp file",
+                Files.exists(file.value()),
+                Matchers.is(true)
+            );
+        }
+    }
+
+    @Test
+    public void createFileInCustomPath() throws Exception {
+        final Path custom = Paths.get(System.getProperty("user.home"));
+        try (final TempFile file = new TempFile(() -> custom, "", "")) {
+            MatcherAssert.assertThat(
+                "Cannot create a temp file at a custom path",
+                file,
+                Matchers.allOf(
+                    new MatcherOf<>(
+                        tmp -> {
+                            return Files.exists(tmp.value());
+                        }
+                    ),
+                    new MatcherOf<>(
+                        tmp -> {
+                            return tmp.value().getParent().equals(custom);
+                        }
+                    )
+                )
+            );
+        }
+    }
+
+    @Test
+    public void closedFileCeasesToExist() throws Exception {
+        final TempFile file = new TempFile();
+        file.close();
+        MatcherAssert.assertThat(
+            "Cannot delete file on close",
+            Files.exists(file.value()),
+            Matchers.is(false)
+        );
+    }
+
+    @Test
+    public void prefix() throws Exception {
```

Comment:
@llorllale Find a better name for this test? What is ``prefix``? Usually, the tests' name are in the format ``doesSomething`` -- so the class being tested can do that particular thing, e.g. ``createsFile``, ``deletesFile`` etc. 

Same for the test ``suffix`` and it would be nice to adapt the other tests above with the format ``doesSomething`` - they are close to it, but not really there.